test/unit/model/validation.test.js

```diff
@@ -273,10 +273,13 @@ describe(Support.getTestDialectTeaser('InstanceValidator'), () => {
     });
 
     before(function() {
-      this.stub = sinon.stub(current, 'query').callsFake(() => {
-        return new Promise(resolve => {
-          resolve([User.build({}), 1]);
-        });
+      this.stub = sinon.stub(current, 'query').callsFake(sql => {
+        // Force to call composeQuery, which calls validation
+        return Promise.resolve(current.dialect.QueryGenerator.composeQuery(sql)).then(() =>
```

Comment:
I think you are fine just calling this fn and then `return Promise.resolve([...])`?